### PR TITLE
fix(core): register invoice with actual lnd pubkey it was created with

### PR DESCRIPTION
**Pull request by vindard**
## Description

This fixes the issue where invoices were registered in our local persistence with a default pubkey regardless of which lnd the invoice was actually created on.

### CODE REVIEW

File: core/api/src/app/wallets/add-invoice-for-wallet.ts [hunk 1]
```diff
@@ -262,8 +261,7 @@ const addInvoice = async ({
   const walletInvoiceBuilder = WalletInvoiceBuilder({
     dealerBtcFromUsd: dealer.getSatsFromCentsForFutureBuy,
     dealerUsdFromBtc: dealer.getCentsFromSatsForFutureBuy,
-    lnRegisterInvoice: (args) =>
-      lndService.registerInvoice({ ...args, sats: toSats(args.btcPaymentAmount.amount) }),
+    lnRegisterInvoice: lndService.registerInvoice,
```
Comment (dolcalmi) [inline core/api/src/app/wallets/add-invoice-for-wallet.ts:265]: Weren't we using sats?

File: core/api/src/domain/bitcoin/lightning/index.types.d.ts [hunk 2]
```diff
@@ -219,7 +215,7 @@ interface ILightningService {
   }): Promise<RawRoute | LightningServiceError>
 
   registerInvoice(
-    args: RegisterInvoiceArgs,
+    args: RegisterInvoiceArgs & { lndsAndPubkeys: LndAndPubkey[] },
```
Comment (dolcalmi) [inline core/api/src/domain/bitcoin/lightning/index.types.d.ts:218]: this is not a good interface update, this method is an abstraction that must not require to pass lnds info or "available nodes" (this must be hidden inside service). Probably is better to use pubkey from RegisteredInvoice

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: core/api/src/domain/bitcoin/lightning/index.types.d.ts (hunk 2 above)
Comment (vindard) [reply core/api/src/domain/bitcoin/lightning/index.types.d.ts]: Ah true, missed this! I got distracted by trying to find a way to test this from outside the service. Fixed now

File: core/api/src/app/wallets/add-invoice-for-wallet.ts (hunk 1 above)
Comment (vindard) [reply core/api/src/app/wallets/add-invoice-for-wallet.ts:264]: We had a new interface that uses `btcPaymentAmount` instead that we hadn't switched over to, so I did the switchover here as well https://github.com/GaloyMoney/galoy/pull/4135/commits/cfd5055875b53eaa0d85ce65f79ea26bb8b3da8a

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**